pkg/providers/core.go

```diff
@@ -81,6 +81,8 @@ func ServiceProviders() di.Option {
 
 		di.Provide(aws.NewDefaultClientFactory, di.As(new(aws.ClientFactory))),
 
+		di.Provide(acl.NewEnterpriseClusterRegistrationAccessListMiddleware),
```

Comment:
The acl checking mechanism seems specific to kafka, I think we could move the middleware and the config out of the pkg/providers/core.go to https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/40fbcaa88a6bda9e82093f9f7774ab6180b562c2/internal/kafka/providers.go

Reply:
thanks. moved it to internal folder